scripts/buildVM.sh

```diff
@@ -55,18 +52,11 @@ while getopts "i:n:r:d:m:a:hv" opt; do
    m )  memSizeMB="$OPTARG" ;;
    n )  vmName="$OPTARG" ;;
    r )  initramfsRecipeName="$OPTARG" ;;
-   v )  verbose_mode=1 ;;
    \?)  print_usage_and_die ;;
    esac
 done
 shift $(($OPTIND - 1))
 
-if [ "$verbose_mode" -eq 0 ] ; then
-    do_silent() { "$@" &>/dev/null; }
```

Comment:
Learning here... Can you explain how redirecting $@ squashes qemu output?

Reply:
The `do_silent()` function here is just calling the function parameters `$@` as a generic shell command, but with a redirect applied.

```bash
# So...
do_silent qemu-system...
# becomes...
qemu-system ... &>/dev/null
# ... within the context of the function
```

`&>/dev/null` is interpreted by the shell as: *within the context of the previous command/function, open a file pointer to `/dev/null` and copy it into **all open file pointers** `&`*

Reply:
Fun fact that I learned recently: the redirect semantics work the same way for command groups and functions, as they do for single commands. So all of the following are exactly equivalent:
```bash
echo some test text 1>/dev/null

{
  echo some test text
} 1>/dev/null

do_silent_1() {
  $@ 1>/dev/null
}
do_silent_1 echo some test text

do_silent_2() {
  $@
} 1>/dev/null
do_silent_2 echo some test text
```

Reply:
Good to know. Thanks for the detailed explanation!